src/crush/CrushWrapper.cc

```diff
@@ -3131,7 +3131,21 @@ class CrushTreeFormattingDumper : public CrushTreeDumper::FormattingDumper {
     f->open_array_section("nodes");
     Parent::dump(f);
     f->close_section();
+
+    // There is no stray bucket whose id is a negative number, so just get
+    // the max_id and iterate from 0 to max_id to dump stray osds.
     f->open_array_section("stray");
+    auto &name_map = this->crush->name_map;
+    int32_t max_id = std::numeric_limits<int32_t>::min();
+    for (auto& item : name_map) {
```

Comment:
nit, might want to put
```c++
int32_t max_id = -1;
if (!name_map.empty()) {
  max_id = name_map.rbegin()->first;
}
```

because
- as you put, all osd id is greater or equal to zero. so `-1` is a safe number as the lower bound of the item id for stray OSDs. and accompanied by the comment, its magic level is decreased (in the sense of  magic number)
- `CrushWrapper::name_map` is an `std::map`, which is in turn an ordered map. so the last element in it has the greatest key.